File: Java/Android Architecture Component/RecyclerView/app/src/main/java/com/contextidea/recycleview/MainActivity.java
package com.contextidea.recycleview;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        recyclerView=(RecyclerView) findViewById(R.id.recycler_view);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.addItemDecoration(new DividerItemDecoration(this,DividerItemDecoration.VERTICAL));
        recyclerView.setAdapter(new RecyclerViewAdapter(this,getData()));

    }

    private ArrayList<SportViewModel> getData(){
        ArrayList<SportViewModel> data=new ArrayList<SportViewModel>();
         SportViewModel model1=new SportViewModel();
         model1.sportName="Football";
         model1.imgSrcId.set(R.drawable.star_outline);
         data.add(model1);
        SportViewModel model2=new SportViewModel();
        model2.sportName="Cricket";
        model2.imgSrcId.set(R.drawable.star_outline);
        data.add(model2);
        SportViewModel model3=new SportViewModel();
        model3.sportName="Busket";
        model3.imgSrcId.set(R.drawable.star_outline);
        data.add(model3);
        return data;
    }
}

File: Kotlin/android-rxjava-retrofit-kotlin-master-crud/RetrofitKotlin/app/src/main/java/com/learn2crack/retrofitkotlin/MainActivity.kt
package com.learn2crack.retrofitkotlin

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.widget.Toast

import io.reactivex.disposables.CompositeDisposable
import io.reactivex.schedulers.Schedulers
import io.reactivex.android.schedulers.AndroidSchedulers

import com.learn2crack.retrofitkotlin.network.RequestInterface
import com.learn2crack.retrofitkotlin.adapter.DataAdapter
import com.learn2crack.retrofitkotlin.model.Post
import com.learn2crack.retrofitkotlin.network.RetrofitClient
import kotlinx.android.synthetic.main.activity_main.*

import kotlinx.android.synthetic.main.content_main.*


class MainActivity : AppCompatActivity(), DataAdapter.Listener {

    private val TAG = MainActivity::class.java.simpleName

    private var mCompositeDisposable: CompositeDisposable? = null

    private var mAndroidArrayList: ArrayList<Post>? = null

    private var mAdapter: DataAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        fab.setOnClickListener {
            startActivity(Intent(this,AddActivity::class.java))
        }

        mCompositeDisposable = CompositeDisposable()

        initRecyclerView()

        loadJSON()

    }

    private fun initRecyclerView() {

        recyclerView1.setHasFixedSize(true)
        val layoutManager : RecyclerView.LayoutManager = LinearLayoutManager(this)
        recyclerView1.layoutManager = layoutManager
    }

    private fun loadJSON() {
        val retrofit=RetrofitClient.instance
        val requestInterface = retrofit.create(RequestInterface::class.java)

        mCompositeDisposable?.add(requestInterface.getData()
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeOn(Schedulers.io())
                .subscribe(this::handleResponse, this::handleError))

    }

    private fun handleResponse(androidList: List<Post>) {

        mAndroidArrayList = ArrayList(androidList)
        mAdapter = DataAdapter(mAndroidArrayList!!, this)

        recyclerView1.adapter = mAdapter
    }

    private fun handleError(error: Throwable) {

        Log.d(TAG, error.localizedMessage)

        Toast.makeText(this, "Error ${error.localizedMessage}", Toast.LENGTH_SHORT).show()
    }

    override fun onItemClick(post: Post) {

        Toast.makeText(this, "${post.title} Clicked !", Toast.LENGTH_SHORT).show()
    }

    override fun onDestroy() {
        super.onDestroy()
        mCompositeDisposable?.clear()
    }
}
